main.go

```diff
@@ -354,17 +354,38 @@ type noEnveloper interface {
 	WithoutEnvelopes() encoding.Serializer
 }
 
+func createJaegerTracer(opts Options, out output) (opentracing.Tracer, io.Closer) {
+	// yab must set the `SynchronousInitialization` flag to indicate that
+	// the Jaeger client must fetch debug credits synchronously. In a
+	// short-lived process like yab, the Jaeger client cannot afford to
+	// postpone the credit request for later in time.
+	//
+	// In the event that no Jaeger agent is found, the client will silently
+	// ignore debug spans. This behavior is no different than past
+	// non-throttling behavior, seeing as no Jaeger agent is available to
+	// receive spans (debug or otherwise). In short, the value of `err` will be
+	// `nil` regardless of whether or not an agent is present and/or fails to
+	// dispense credits to the client synchronously.
+	tracer, closer, err := jaeger_config.Configuration{
+		ServiceName: opts.TOpts.CallerName,
+		Throttler: &jaeger_config.ThrottlerConfig{
+			SynchronousInitialization: true,
+		},
+	}.NewTracer()
+	if err != nil {
+		out.Fatalf("Failed to create Jaeger tracer: %s", err.Error())
+	}
+	return tracer, closer
+}
+
 func getTracer(opts Options, out output) (opentracing.Tracer, io.Closer) {
-	var (
-		tracer opentracing.Tracer = opentracing.NoopTracer{}
-		closer io.Closer
-	)
 	if opts.TOpts.Jaeger && !opts.TOpts.NoJaeger {
-		tracer, closer = jaeger.NewTracer(opts.TOpts.CallerName, jaeger.NewConstSampler(true), jaeger.NewNullReporter())
```

Comment:
@prashantv why was yab using a null reporter? Do you have a specific need to omit the root span?

Reply:
I have just changed this based on @yurishkuro's advice. Unless you mind, we might as well do this all at once.

Reply:
I was going to suggest not making this discussion a part of this PR, but move to #234.

Reply:
I'm going to undo this and potentially open a new PR about this.